src/masking/mod.rs

```diff
@@ -1 +1,91 @@
+use core::mem::MaybeUninit;
+use core::num::{NonZeroUsize, Wrapping};
+
+use crate::capacity::MaskingCapacity;
+use crate::storage::{PartialStorage, Storage};
+
 mod tests;
+
+pub struct MaskingRingBuffer<S: PartialStorage<Capacity = MaskingCapacity>> {
+    /// The underlying storage
+    storage: MaybeUninit<S>,
```

Comment:
Can this be moved to the last field?  That way we can allow for `?Sized` storages in the future.